app/metal-controller-manager/api/v1alpha1/serverclass_filter.go

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1
+
+import "sort"
+
+// FilterAcceptedServers returns a new slice of Servers that are accepted and qualify.
+//
+// Returned Servers are always sorted by name for stable results.
+func FilterAcceptedServers(servers []Server, q Qualifiers) []Server {
```

Comment:
Converted the previous version of this code to use pure functions instead of _both_ mutating an internal state and returning it